codegen-core/common-test-models/rest-json-extras-2310.smithy

```diff
@@ -0,0 +1,36 @@
+$version: "1.0"
```

Comment:
We should write all new tests in 2.0 syntax. But since the tests in this file are copied from the ones upstream in Smithy (and they'll go away in the next Smithy release), this is OK, I just made the same modifications that I did upstream: https://github.com/smithy-lang/smithy/pull/2310